dali/pipeline/data/types.h

```diff
@@ -436,8 +439,9 @@ class DLL_PUBLIC TypeTable {
   }
 
   template <typename T>
-  DLL_PUBLIC static const TypeInfo& GetTypeInfoFromStatic() {
-    return GetTypeInfo(GetTypeID<T>());
+  DLL_PUBLIC static const TypeInfo& GetTypeInfo() {
+    static const TypeInfo &type_info = GetTypeInfo(GetTypeID<T>());
+    return type_info;
```

Comment:
Why can't we just return the reference directly?

Reply:
Because we want to get it only once in the lifetime of the process.